src/types/verifiableCredentials/index.ts

```diff
@@ -0,0 +1,51 @@
+export interface Credential {
+  context: string[];
+  id: string;
+  type: string[];
+  issuer: string | Issuer;
+  issuanceDate: Date;
+  expirationDate?: Date;
+  credentialSubject: Subject;
+  credentialStatus?: Status;
+}
+
+export interface VerifiableCredential extends Credential {
+  proof?: Proof[];
+}
+
+export interface Presentation {
+  context: string[];
+  id?: string;
+  type: string[];
+  verifiableCredential?: VerifiableCredential[];
+  holder?: string;
+}
+
+export interface VerifiablePresentation extends Presentation {
+  proof?: Proof[];
+}
+
+export interface Issuer {
+  id?: string;
+}
+
+export interface Subject {
+  id?: string;
+  claims: Record<string, ClaimValue>;
+}
+
+export interface Status {
+  id: string;
+  type: string;
+}
+
+export type ClaimValue = string | ClaimValue[] | { [key: string]: ClaimValue };
```

Comment:
```suggestion
export type ClaimValue = string | ClaimValue[] | Record<string, ClaimValue>;
```

Reply:
Using the Index Signature avoids the Circular Type Dependency which arises from using a Record. I'm not sure how to get around this without creating a separate type signature. Do you have any suggestions?

Reply:
Ok, that's fine. Don't know why Record doesn't work here (it's just an alias).

Reply:
Yes thats what I thought, I'm curious as well